pkg/sql/sem/tree/datum.go

```diff
@@ -2094,11 +2108,25 @@ type DTimestamp struct {
 }
 
 // MakeDTimestamp creates a DTimestamp with specified precision.
-func MakeDTimestamp(t time.Time, precision time.Duration) *DTimestamp {
-	return &DTimestamp{Time: t.Round(precision)}
+func MakeDTimestamp(t time.Time, precision time.Duration) (*DTimestamp, error) {
+	ret := t.Round(precision)
+	if ret.After(MaxSupportedTime) || ret.Before(MinSupportedTime) {
+		return nil, errors.Newf("timestamp %q exceeds supported timestamp bounds", ret.Format(time.RFC3339))
+	}
+	return &DTimestamp{Time: ret}, nil
 }
 
-var dMinTimestamp = &DTimestamp{}
+// MustDTimestamp wraps MakeDTimestamp but panics if there is an error.
+// This is intended for testing applications only.
+func MustDTimestamp(t time.Time, precision time.Duration) *DTimestamp {
```

Comment:
`MustDTimestamp` feels awkward. I guess we already have a function `MustBeDTimestamp`, so that's already taken. How about `MustBeValidDTimestamp`?

Reply:
I've done `MustMakeDTimestamp`.